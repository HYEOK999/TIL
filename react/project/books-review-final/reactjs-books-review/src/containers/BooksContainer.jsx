// import React from 'react';

import { connect } from 'react-redux';
import Books from '../components/Books';
import { startBooksSaga, startDeleteBookSaga } from '../redux/modules/books';
// import { useCallback } from 'react';

// const BooksContainer = props => {
//   const books = useSelector(state => state.books.books);
//   const loading = useSelector(state => state.books.loading);
//   const error = useSelector(state => state.books.error);

//   const dispatch = useDispatch();

//   const getBooks = useCallback(() => {
//     dispatch(startBooksSaga());
//   }, [dispatch]);

//   return (
//     <Books
//       {...props}
//       books={books}
//       loading={loading}
//       error={error}
//       getBooks={getBooks}
//     />
//   );
// };

export default connect(
  state => ({
    books: state.books.books,
    loading: state.books.loading,
    error: state.books.error,
  }),
  dispatch => ({
    getBooks: () => {
      dispatch(startBooksSaga());
    },
    deleteBook: bookId => {
      dispatch(startDeleteBookSaga(bookId));
    },
  }),
)(Books);
