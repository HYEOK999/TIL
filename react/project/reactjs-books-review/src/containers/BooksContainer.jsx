import React, { useCallback } from 'react';
import { connect, useSelector, useDispatch } from 'react-redux';
import Books from '../components/Books';
import {
  getBooks as getBooksAction,
  startBooksSaga,
} from '../redux/modules/books';

// const mapStateToProps = state => ({
//   books: state.books.books,
//   token: state.auth.token,
//   loading: state.books.loading,
//   error: state.books.error,
// });

// const mapDispatchToProps = dispatch => ({
//   getBooks: async token => {
//     dispatch(getBooks(token));
//   },
// });

// export default connect(mapStateToProps, mapDispatchToProps)(Books);
const BookContainer = props => {
  // const token = useSelector(state => state.auth.token);
  const books = useSelector(state => state.books.books);
  const loading = useSelector(state => state.books.loading);
  const error = useSelector(state => state.books.error);
  const dispatch = useDispatch();

  const getBooks = useCallback(() => {
    // dispatch(getBooksAction());
    // setInterval(() => {
    //   dispatch(startBooksSaga());
    //   dispatch(startBooksSaga());
    //   dispatch(startBooksSaga());
    //   dispatch(startBooksSaga());
    //   dispatch(startBooksSaga());
    // }, 100)
    dispatch(startBooksSaga());
  }, [dispatch]);

  return (
    <Books
      {...props}
      books={books}
      loading={loading}
      error={error}
      getBooks={getBooks}
    />
  );
};

export default BookContainer;
