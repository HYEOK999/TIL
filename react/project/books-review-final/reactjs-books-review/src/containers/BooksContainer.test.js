import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import BooksContainer from './BooksContainer';

Enzyme.configure({ adapter: new Adapter() });

describe('BooksContainer', () => {
  it('renders properly', () => {
    const createStore = configureMockStore();

    const booksMock = [
      {
        bookId: 1,
        ownerId: '7d26db27-168c-4c6a-bd9a-9e20677b60b8',
        title: '모던 자바스크립트 입문',
        message: '모던하군요',
      },
      {
        bookId: 2,
        ownerId: '7d26db27-168c-4c6a-bd9a-9e20677b60b8',
        title: '책 Mock',
        message: '메세지 Mock',
      },
      {
        bookId: 3,
        ownerId: '7d26db27-168c-4c6a-bd9a-9e20677b60b8',
        title: '책 Mock 222',
        message: '메세지 Mock 222',
      },
    ];

    const store = createStore({
      books: {
        books: booksMock,
        loading: true,
        error: null,
      },
      auth: {
        token: '일단 모르겠다',
        loading: false,
        error: null,
      },
    });

    const component = mount(<BooksContainer store={store} />);
    expect(component).toMatchSnapshot();
  });
});
