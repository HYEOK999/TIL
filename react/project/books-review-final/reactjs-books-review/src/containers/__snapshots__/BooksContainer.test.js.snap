// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksContainer renders properly 1`] = `
<Connect(Books)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Books
    books={
      Array [
        Object {
          "bookId": 1,
          "message": "모던하군요",
          "ownerId": "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
          "title": "모던 자바스크립트 입문",
        },
        Object {
          "bookId": 2,
          "message": "메세지 Mock",
          "ownerId": "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
          "title": "책 Mock",
        },
        Object {
          "bookId": 3,
          "message": "메세지 Mock 222",
          "ownerId": "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
          "title": "책 Mock 222",
        },
      ]
    }
    error={null}
    getBooks={[Function]}
    loading={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <p>
      로딩 중...
    </p>
    <ul>
      <li
        key="1"
      >
        모던 자바스크립트 입문
      </li>
      <li
        key="2"
      >
        책 Mock
      </li>
      <li
        key="3"
      >
        책 Mock 222
      </li>
    </ul>
  </Books>
</Connect(Books)>
`;
